src/libdrakvuf/win.c

```diff
@@ -244,6 +244,41 @@ static bool find_kernbase(drakvuf_t drakvuf)
     return 1;
 }
 
+addr_t win_get_function_argument(drakvuf_t drakvuf, drakvuf_trap_info_t* info, int narg)
+{
+    page_mode_t pm = drakvuf_get_page_mode(drakvuf);
+    if (pm == VMI_PM_IA32E)
+    {
+        switch (narg)
+        {
+            case 1:
+                return info->regs->rcx;
+            case 2:
+                return info->regs->rdx;
+            case 3:
+                return info->regs->r8;
+            case 4:
+                return info->regs->r9;
+        }
+    }
+
+    access_context_t ctx =
+    {
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3,
+        .addr = info->regs->rsp + narg * drakvuf_get_address_width(drakvuf),
+    };
+
+    addr_t addr;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
```

Comment:
Within `libdrakvuf` itself there is no need for locking, that's just for plugins.

Reply:
Done.